pkgs/by-name/dd/ddns-updater/package.nix

```diff
@@ -0,0 +1,32 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+}:
+buildGoModule rec {
+  pname = "ddns-updater";
+  version = "2.5.0";
+
+  src = fetchFromGitHub {
+    owner = "qdm12";
+    repo = "ddns-updater";
+    rev = "v${version}";
+    hash = "sha256-DQv6WG5qvwMCLU2yrR/Fa/81C1EAFq2lDMdo0+C8ono=";
+  };
+
+  vendorHash = "sha256-hRcmaz5iG9sWRzzIsMTW6qchlP84jhwOD3PjoeWIW5g=";
+
+  ldflags = [
+    "-s"
+    "-w"
+  ];
+
+  meta = with lib; {
+    description = "Container to update DNS records periodically with WebUI for many DNS providers";
+    homepage = "https://github.com/qdm12/ddns-updater";
+    license = licenses.mit;
+    maintainers = with maintainers; [delliott];
+    mainProgram = "updater";
+    platforms = platforms.linux;
```

Comment:
The most recent release (2.6.0) provides darwin binaries, might be worth looking into.